src/main/java/de/adesso/kicker/notification/NotificationService.java

```diff
@@ -0,0 +1,84 @@
+import de.adesso.kicker.notification.MatchVerificationRequest.VerifyMatchService;
+import de.adesso.kicker.notification.Message.Message;
+import de.adesso.kicker.notification.Message.MessageType;
+import de.adesso.kicker.notification.Message.SendMessageService;
+import de.adesso.kicker.notification.exception.NotificationNotExistingException;
+import de.adesso.kicker.notification.exception.WrongReceiverException;
+import de.adesso.kicker.user.User;
+import de.adesso.kicker.user.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class NotificationService {
+
+    private NotificationRepository notificationRepository;
+
+    private VerifyMatchService verifyMatchService;
+
+    private SendMessageService sendMessageService;
+
+    private UserService userService;
+
+    @Autowired
+    public NotificationService(NotificationRepository notificationRepository, VerifyMatchService verifyMatchService,
+            UserService userService, SendMessageService sendMessageService) {
+        this.notificationRepository = notificationRepository;
+        this.verifyMatchService = verifyMatchService;
+        this.sendMessageService = sendMessageService;
+        this.userService = userService;
+    }
+
+    public void acceptNotification(long notificationId) {
+        Notification notification = checkNotificationExists(notificationId);
+        checkWrongReceiver(notification);
+
+        switch (notification.getType()) {
+        case MESSAGE:
+            sendMessageService.deleteMessage((Message) notification);
+            break;
+        case MATCH_VERIFICATION:
+            verifyMatchService.acceptRequest((MatchVerificationRequest) notification);
+            break;
+        }
+    }
+
+    public void declineNotification(long notificationId) {
+        Notification notification = checkNotificationExists(notificationId);
+        checkWrongReceiver(notification);
+
+        switch (notification.getType()) {
+        case MESSAGE:
+            sendMessageService.deleteMessage((Message) notification);
+            break;
+        case MATCH_VERIFICATION:
+            List<User> users = verifyMatchService.declineRequest((MatchVerificationRequest) notification);
+            User sender = userService.getLoggedInUser();
+            for (User user : users) {
+                sendMessageService.sendMessage(sender, user, MessageType.MESSAGE_DECLINED);
+            }
+            break;
+        }
+    }
+
+    public List<Notification> getNotificationsByReceiver(User receiver) {
+        return notificationRepository.findAllByReceiver(receiver);
+    }
+
+    public void checkWrongReceiver(Notification notification) {
+        if (notification.getReceiver() != userService.getLoggedInUser()) {
+            throw new WrongReceiverException();
+        }
+    }
+
+    public Notification checkNotificationExists(Long notificationId) {
```

Comment:
Still not quite what I meant. Look at how it is done in `MatchService#getMatchById`

Reply:
getMatchById is almost what i had before enhanceing the function with orElseThrow - what you suggested to do. should i really take this step back?

Reply:
The version of `getMatchById` in your code is outdated.
Here is what I meant
````java
public getMatchById(long id) {
    notificationRepository.findById(id).orElseThrow(SomeException::new);
}